tracer/src/Datadog.Trace/Agent/StatsBuffer.cs

```diff
@@ -57,9 +57,15 @@ public void Reset()
             Start = DateTimeOffset.UtcNow.ToUnixTimeNanoseconds();
         }
 
-        public void Serialize(Stream stream, long bucketDuration)
+        public void Serialize(Stream stream, long bucketDuration, bool propagateProcessTags)
         {
-            MessagePackBinary.WriteMapHeader(stream, 8);
+            var count = 8;
+            if (propagateProcessTags && !string.IsNullOrEmpty(ProcessTags.SerializedTags))
```

Comment:
nit: static access ca be relatively slow, particularly in .NET Framework, I would cache ProcessTags.SerializedTags to a variable instead of doing 3 accesses